helperFunctions/SalesforceMessageHelpers.ts

```diff
@@ -7,12 +7,34 @@ import { ErrorLogs } from '../enum/ErrorLogs';
 import { getAppSettingValue } from '../lib/Settings';
 import { agentTypingListener, removeAgentTypingListener } from './AgentTypingHelper';
 import { sendLCMessage } from './LivechatMessageHelpers';
+import { retrievePersistentTokens } from './PersistenceHelpers';
 
-export async function messageFilter(app: IApp, modify: IModify, read: IRead, messageRoom: IRoom, LcAgent: IUser, messageArray: any) {
+export async function messageFilter(app: IApp, modify: IModify, read: IRead, messageRoom: IRoom, LcAgent: IUser, messageArray: any, assoc, persistence) {
 	try {
 		messageArray.forEach(async (i) => {
 			const type = i.type;
 			switch (type) {
+				case 'ChatTransferred':
+					const transferMessage = i.message;
+					const chasitorIdleTimeout = transferMessage.chasitorIdleTimeout || false;
+					const sneakPeekEnabled = transferMessage.sneakPeekEnabled;
+					const { id, persisantAffinity, persistantKey } = await retrievePersistentTokens(read, assoc);
+					const salesforceAgentName = transferMessage.name;
+
+					await persistence.updateByAssociation(
+						assoc,
+						{
+							id,
+							affinityToken: persisantAffinity,
+							key: persistantKey,
+							chasitorIdleTimeout,
+							sneakPeekEnabled,
+							salesforceAgentName,
+						},
+						true,
+					);
```

Comment:
@csmith14 We should use `updatePersistentData` here. There are other properties that we are not covering here. And updating by this object will reset it and we will lose those properties. i.e, `isIdleSessionTimerScheduled`

https://github.com/WideChat/Apps.Salesforce.LiveAgent/blob/master/helperFunctions/PersistenceHelpers.ts#L64...L78